fp/api/views.py

```diff
@@ -0,0 +1,56 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from .models import Docket, Document
+from .serializers import DocketSerializer, DocumentSerializer
+
+# Create your views here.
+class DocketView(GenericAPIView):
+    queryset = Docket.objects.all()
+    serializer_class = DocketSerializer
+
+    def get(self, request: Request):
+        queryset = self.get_queryset()
+        docket_no = request.data.get("docket_no")
+        if docket_no != None:
+            docket = get_object_or_404(queryset, pk=docket_no)
+            serializer = self.get_serializer(docket)
+            return Response(serializer.data)
+        serializer = self.get_serializer(queryset, many=True)
+        return  Response(serializer.data)
+    
+    
+    # {'docket_no': '23-09001',
+    # 'date_filled': '09/01/2023',
+    # 'description': 'Informational Report of Sierra Pacific Power Company d/b/a NV Energy concerning its natural gas resource planning activities.'}
+    def post(self, request: Request):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
No if statement required here because if this fails then it will raise exception, remove if statement

Reply:
okay!

Reply:
commit 5890b17fd326a4dc1dbd09d4193ee5a3b17304fa resolves this